projects/jump_ci/testing/tunnelling.py

```
@@ -0,0 +1,205 @@
+    -L {local_host_port}:{bastion_host}:{bastion_host_port} \
+    -N"
+
+    if verbose:
+        logging.info(cmd)
+
+    RETRIES = 150
+    DELAY = 5
+    recreate_count_down = RECREATED_TUNNEL_COUNT_DOWN = 5 # recreate the SSH tunnel every 5 attempts
+
+    def create_tunnel():
+        _proc = subprocess.Popen(cmd, shell=True)
```

Ruff (0.15.5)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)